src/components/commons/Buttons.tsx

```diff
@@ -1,24 +1,147 @@
 import { PropsWithChildren } from "react";
 import classNames from "classnames";
 
-/**
- * Container implementation that is part of the design language.
- */
-export function Button(
-  props: PropsWithChildren<{ className?: string; text: string }>
-): JSX.Element {
+export function Button({
+  buttonText,
+  className,
+  disabled,
+  href,
+  onClick,
+  isCTAButton = true,
+}: {
+  buttonText: string;
+  className?: string;
+  disabled?: boolean;
+  href?: string;
+  onClick?: () => void;
+  isCTAButton?: boolean;
+}) {
+  if (href) {
+    return (
+      <a
+        href={href}
+        className={classNames({ "pointer-events-none": disabled })}
+      >
+        <ButtonElement
+          buttonText={buttonText}
+          className={className}
+          disabled={disabled}
+          onClick={onClick}
+          isCTAButton={isCTAButton}
+        />
+      </a>
+    );
+  }
+  return (
+    <ButtonElement
+      buttonText={buttonText}
+      className={className}
+      disabled={disabled}
+      onClick={onClick}
+      isCTAButton={isCTAButton}
+    />
+  );
+}
+
+function ButtonElement({
+  buttonText,
+  className,
+  disabled,
+  onClick,
+  isCTAButton = true,
+}: {
+  buttonText: string;
+  className?: string;
+  disabled?: boolean;
+  onClick?: () => void;
+  isCTAButton?: boolean;
+}) {
   return (
     <button
+      type="button"
+      onClick={onClick}
+      disabled={disabled}
       className={classNames(
-        "button rounded border border-primary-500 py-2 px-6 font-medium text-white bg-primary-500 hover:bg-primary-600",
-        props.className
+        className ?? "py-4 px-28",
+        "flex items-center justify-center rounded-[92px] font-bold disabled:opacity-30 disabled:pointer-events-none",
+        isCTAButton
+          ? "bg-dark-1000 text-dark-100 hover:text-dark-00 active:text-dark-00 hover:cta-on-hover active:cta-on-press"
```

Comment:
what is:
- `cta-on-hover` 
- `cta-on-press`
- `on-button-hover`
- `on-button-press`

Reply:
these styles are in globals.css as the buttons on hover and on press are linear gradients background

Reply:
sorry saw the changes on dsm will update

Reply:
updated and added `<LinkButton />`